src/plasma/plasma_client.cc

```diff
@@ -465,13 +469,67 @@ void plasma_contains(PlasmaConnection *conn, ObjectID obj_id, int *has_object) {
   }
 }
 
+void compute_block_hash(const unsigned char *data, int64_t nbytes, uint64_t *hash) {
+  XXH64_state_t hash_state;
+  XXH64_reset(&hash_state, XXH64_DEFAULT_SEED);
+  XXH64_update(&hash_state, data, nbytes);
+  *hash = XXH64_digest(&hash_state);
+}
+
+bool compute_object_hash_parallel(XXH64_state_t *hash_state,
+                                  const unsigned char *data,
+                                  int64_t nbytes) {
+#ifndef NUMTHREADS
+#define NUMTHREADS 8
+#endif
+  const uint64_t numthreads = NUMTHREADS;
+  uint64_t threadhash[numthreads+2];
+  //CHECK(numthreads > 0);
+  const uint64_t blocksz = 64; // cache block alignment (alternative: page size)
+  // Calculate the first and last aligned positions in the data stream.
+  unsigned char *databp = (unsigned char *)(((uint64_t)data + blocksz-1) & ~(blocksz-1));
```

Comment:
there are a few ways this could be simplified, like doing a shift right, shift left by the alignment bitwidth, but it may get weird on different architectures, if you are not careful. Because the shift can be with/without sign bit, with/without carry. It would also require us to add a variable that derives the alignment bitwidth of the specified block size (i.e. a log base 2 of blocksz).